models/plays_by_pageview.sql

```diff
@@ -0,0 +1,140 @@
+{{ 
+  config(
+    materialized='table',
+    unique_key = 'play_id',
+    sort = 'start_tstamp',
+    dist = 'play_id'
+  )
+}}
+
+with prep as (
+
+  select
+    {{ dbt_utils.hash('i.page_view_id||i.media_id') }} play_id,
+    i.page_view_id,
+    i.media_id,
+    max(i.domain_sessionid) as domain_sessionid,
+    i.domain_userid,
+    i.duration,
+    i.title,
+    i.media_type,
+    i.media_player_type,
+    i.page_referrer,
+    i.content_url,
+    i.geo_region_name,
+    i.br_name,
+    i.dvce_type,
+    i.os_name,
+    i.os_timezone,
+    coalesce(sum(p.weight_rate * i.duration), 0)::int as play_time_sec_estimated,
+    min(start_tstamp) start_tstamp,
+    sum(case when i.event_type in ('seek', 'seeked') then 1 else 0 end) as seeks,
+    max(case 
+          when i.event_type = 'ended' then 100 
+          when i.percent_progress = 0 then NULL 
+          else i.percent_progress 
+        end) as last_percent_progress,
+    max(i.end_tstamp) end_tstamp,
+    sum(i.play_time_sec_amended) as play_time_sec,
+    sum(case when i.muted then i.play_time_sec_amended else 0 end) as play_time_sec_muted,
+    coalesce(sum(i.playback_rate * i.play_time_sec) / nullif(sum(i.play_time_sec), 0), 0) as avg_playback_rate,
```

Comment:
You are right, let's leave them as nulls in case there is no play time. I also removed the coalesce from the subsequent line, the retention rate, as there should always be a duration and if not, the null value will indicate there is a data issue.